src/application/selector/xml/XmlResultFormatter.js

```diff
@@ -0,0 +1,28 @@
+class XmlResultFormatter {
+  static toString(results) {
+    if (Array.isArray(results)) {
+      switch (results.length) {
+        case 0:
+          return null;
+        case 1:
+          return results[0].toString();
+        default:
+          return this.__wrapCollections(results.map(el => this.__wrapSingleNode(el)));
```

Comment:
```suggestion
          return this.__wrapCollections(results.map(this.__wrapSingleNode));
```
Would this shorter version work?